### PR TITLE
[JETI-7, JETI-4, JETI-3] Upgrade to support @tableland/sdk v4, add publishing, and create tests

### PR DESCRIPTION
**Pull request by awmuncy**
Upgrade to support @tableland/sdk v4.

Add publishing: We have added the necessary configuration and scripts to enable seamless publishing of the library to NPM.

Create tests: Created a few basic tests which can be built on, and include mocks for the ipfs-http lib.

Switch from Jest to Mocha: In order to maintain consistency across our JavaScript repositories, I switched the testing framework from Jest to Mocha.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,4 +2,4 @@
 # the repo. Unless a later match takes precedence.
 # By convention in @tablelandnetwork repos, the first listed
 # CODEOWNER is the repo DRI (directly responsible individual)
-*       @awmuncy
+*       @awmuncy @njwags @carsonfarmer
```
Comment (joewagner) [inline .github/CODEOWNERS:5]: my github use is @joewagner

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      - name: Conditional
+        id: cond
+        uses: haya14busa/action-cond@v1
+        with:
+          cond: ${{ contains(github.event.inputs.release_version, '-') }}
+          if_true: "next"
+          if_false: "latest"
+
+      - name: Publish 📦
+        id: publish
```
Comment (joewagner) [inline .github/workflows/publish.yml:54]: Does this do the build before publishing somehow? I think the SDK uses a `prepublishOnly` script, but I don't see that in the package.json for this package.

**Review by joewagner — COMMENTED** [review]
Left a few comments inline.  Should we setup the standard `staging` branch and point this PR at it?

**Comment by awmuncy** [comment]
> Left a few comments inline. Should we setup the standard `staging` branch and point this PR at it?

Done, set up, added branch protection rules, and pointed this PR at it.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (awmuncy) [reply .github/workflows/publish.yml:54]: Oh right, I think it was being published manually before. I've added the prepublish step and pointed it at "build:all"

File: package.json [hunk 3]
```diff
@@ -37,65 +37,40 @@
...
     "docs:html": "typedoc --entryPoints src/main.ts --plugin none",
     "build:esm": "tsc -p tsconfig.json",
     "build:cjs": "tsc -p tsconfig.cjs.json",
-    "build:esbuild": "esbuild src/main.ts --bundle --minify --sourcemap=external --outfile=dist/esbuild/main.js --format=esm  && node ./node_modules/add-js-extension/dist/bin.js ./dist/esm --once",
+    "build:esbuild": "esbuild src/main.ts --bundle --minify --external:node:* --sourcemap=external --outfile=dist/esbuild/main.js --format=esm  && node ./node_modules/add-js-extension/dist/bin.js ./dist/esm --once",
     "build:dev": "esbuild src/main.ts --bundle --outfile=dist/esbuild/main.js --format=esm --sourcemap=inline",
     "build:watch": "esbuild src/main.ts --bundle --watch --outfile=dist/esbuild/main.js --format=esm && node ./node_modules/add-js-extension/dist/bin.js ./dist/esm --once",
-    "build:all": "npm run build:esm && npm run build:cjs && npm run build:esbuild && ./fixup.sh"
+    "build:all": "npm run build:esm && npm run build:cjs && npm run build:esbuild && ./fixup.sh",
+    "prepublishOnly": "build:all"
```
Comment (joewagner) [inline package.json:53]: I think this will need to be `"prepublishOnly": "npm run build:all".  You can see if it works by doing `npm publish --dry-run`

**Comment by joewagner** [comment]
Looking good! Only last nitpicks are I think the prepublishOnly value has to include `npm run`, but I'm not sure.
Also in our other repos we leave the [package.json version](https://github.com/tablelandnetwork/jeti/blob/main/package.json#L3) should be "0.0.0".  The publish action will update it while running.

**Comment by awmuncy** [comment]
> 

Good point. Updated the version `0.0.0` and add the `npm run` to it.

**Review by joewagner — APPROVED** [review]
Looks good!

**Pull request merged**